src/main/kotlin/com/hypto/iam/server/service/TokenService.kt

```diff
@@ -166,6 +179,10 @@ object MasterKeyCache : KoinComponent {
     }
 
     suspend fun getKey(id: String): MasterKey {
+        if (!::cacheRefreshTime.isInitialized || shouldRefreshCache()) {
```

Comment:
Why do we need a refresh time for cache ? Can we just store the public keys in memory after fetching the keys from db and use that instead always ?

Reply:
If in case the public keys status is changed to EXPIRED, the cache should get updated to reflect the changes.